pkg/ui/v1alpha2/backend.go

```diff
@@ -0,0 +1,493 @@
+	conn, err := grpc.Dial(common_v1alpha2.ManagerAddr, grpc.WithInsecure())
+	if err != nil {
+		log.Printf("Dial to GRPC failed: %v", err)
+		return nil, nil
+	}
+	c := api_pb_v1alpha2.NewManagerClient(conn)
+	return conn, c
+}
+
+func (k *KatibUIHandler) FetchHPJobs(w http.ResponseWriter, r *http.Request) {
+	//enableCors(&w)
+
+	jobs := make([]JobView, 0)
+
+	el, err := k.katibClient.GetExperimentList()
+	if err != nil {
+		log.Printf("GetExperimentList for HP failed: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	for _, experiment := range el.Items {
+		if experiment.Spec.Parameters != nil {
+			experimentLastCondition, err := experiment.GetLastConditionType()
+			if err != nil {
+				log.Printf("GetLastConditionType for HP failed: %v", err)
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+			jobs = append(jobs, JobView{
+				Name:   experiment.Name,
+				Status: string(experimentLastCondition),
+			})
+		}
+	}
+
+	response, err := json.Marshal(jobs)
+	if err != nil {
+		log.Printf("Marshal HP jobs failed: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.Write(response)
+
+}
+
+func (k *KatibUIHandler) FetchNASJobs(w http.ResponseWriter, r *http.Request) {
+	//enableCors(&w)
+
+	jobs := make([]JobView, 0)
+
+	el, err := k.katibClient.GetExperimentList()
+	if err != nil {
+		log.Printf("GetExperimentList for NAS failed: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	for _, experiment := range el.Items {
+		if experiment.Spec.NasConfig != nil {
+			experimentLastCondition, err := experiment.GetLastConditionType()
+			if err != nil {
+				log.Printf("GetLastConditionType for HP failed: %v", err)
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+			jobs = append(jobs, JobView{
+				Name:   experiment.Name,
+				Status: string(experimentLastCondition),
+			})
+		}
+	}
+
+	response, err := json.Marshal(jobs)
+	if err != nil {
+		log.Printf("Marshal NAS jobs failed: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.Write(response)
+
+}
+
+func (k *KatibUIHandler) SubmitYamlJob(w http.ResponseWriter, r *http.Request) {
+	//enableCors(&w)
+	var data map[string]interface{}
+
+	json.NewDecoder(r.Body).Decode(&data)
+
+	job := experimentv1alpha2.Experiment{}
+	if yamlContent, ok := data["yaml"].(string); ok {
+		err := yaml.Unmarshal([]byte(yamlContent), &job)
+		if err != nil {
+			log.Printf("Unmarshal YAML content failed: %v", err)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		err = k.katibClient.CreateExperiment(&job)
+		if err != nil {
+			log.Printf("CreateExperiment from YAML failed: %v", err)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+
+}
+
+func (k *KatibUIHandler) SubmitHPJob(w http.ResponseWriter, r *http.Request) {
+	//enableCors(&w)
```

Comment:
it seems there is no difference between SubmitNASJob and SubmitHPJob

Reply:
Delete one